sdk/eventhub/Azure.Messaging.EventHubs/tests/Consumer/EventHubConsumerClientTests.cs

```diff
@@ -2310,15 +2310,19 @@ private class PublishingTransportConsumerMock : TransportConsumer
                 }
             }
 
-            public override Task<IEnumerable<EventData>> ReceiveAsync(int maximumMessageCount, TimeSpan? maximumWaitTime, CancellationToken cancellationToken)
+            public override async Task<IEnumerable<EventData>> ReceiveAsync(int maximumMessageCount, TimeSpan? maximumWaitTime, CancellationToken cancellationToken)
             {
                 var stopWatch = Stopwatch.StartNew();
                 PublishDelayCallback?.Invoke();
                 stopWatch.Stop();
 
                 if (((maximumWaitTime.HasValue) && (stopWatch.Elapsed >= maximumWaitTime)) || (PublishIndex >= EventsToPublish.Count))
                 {
-                    return Task.FromResult(Enumerable.Empty<EventData>());
+                    // Delay execution in this path to prevent a tight loop, starving other Tasks.
+
+                    await Task.Delay(100);
```

Comment:
Interesting.... this hadn't occurred to me at all.   Great catch!   

Should we consider `ConfigureAwait` here to further reduce contention?

Reply:
Yes, good catch on my catch 😄

Reply:
Fixed, and ready to merge.